### DISCUSSION TITLE
Set Up: single app vs multiple

### DISCUSSION THREAD

**Discussion by redblindbananaz** [The-Digital-Dilemma/GetIT#5] [General]
### What would be the preference for the structure of the project in terms of app?

By separating user management/authentication from course management, we can keep your codebase organised and maintain a clear separation of concerns. Each app can have its own models, views, templates, and URLs specific to its functionality.

- User Management/Auth App: This app will handle user registration, login, authentication, and profile management.
- Course Management App: This app will focus on managing courses, chapters, lessons, and other related functionalities.

The other option is to have one application to manage the user and the courses:

- It simplifies the overall project structure, reduces code duplication, and makes it easier to navigate and understand the codebase.

Please leave a comment about your preference

**Comment by redblindbananaz** [comment]
I personnel prefer having that separation of concern.

**Reply by makikamiya** [reply]
I agree. Having defined responsibilities that are separated into different apps will, I think, make the project's structure more modular and easier to understand and work with. I had a single app in my first assignment and it got very messy/tricky keeping track of where different responsibilities lay.

As for the apps, themselves, I think what you've indicated, Sebastien, are the same as what I would start with for the basic phase (perhaps with the addition of a Home app):

- **Home app** - Responsible for the landing page.
- **Users app** - Responsible for user administration.
- **Courses app** - Responsible for managing courses and their content.

**Reply by redblindbananaz** [reply]
I did a little more research on that and I agree with you with the home app. Home app could host some widget and component such as showcasing popular courses, highlighting upcoming events, or displaying statistics from our data base (ex: number of users, number of lessons available, just as for demonstration of capabilities) or event local time? Which could be some features that can be implemented.
